pkg/auth/authenticator/interfaces.go

```diff
@@ -13,3 +15,28 @@ type Assertion interface {
 type Client interface {
 	AuthenticateClient(client api.Client) (user.Info, bool, error)
 }
+
+type OAuthTokenUserValidator interface {
```

Comment:
Should be called OAuthTokenValidator
The fact one of the validators checks users does not change the generic nature of the validator.